service/matching/matching_engine.go

```diff
@@ -272,6 +272,8 @@ func (e *matchingEngineImpl) Stop() {
 	_ = e.serviceResolver.RemoveListener(e.listenerKey())
 	close(e.membershipChangedCh)
 
+	e.nexusEndpointClient.notifyOwnershipChanged(false)
```

Comment:
It would be nice to be able to wait on the goroutine here for graceful shutdown. Ideally there won't be any background routines when this method returns. Note that there could be multiple running in this use case in extreme cases.